src/foam/java/refinements.js

```diff
@@ -239,7 +239,7 @@ foam.CLASS({
       // Special merging behaviour for args.
       var argCount = Math.max(this.args.length, child.args.length)
       for ( var i = 0 ; i < argCount ; i++ ) {
-        result.args[i] = this.args[i].clone().copyFrom(child.args[i]);
+        result.args[i] = (this.args.length <= i) ? child.args[i] : this.args[i].clone().copyFrom(child.args[i]);
       }
```

Comment:
Sure, the second loop can even be removed I believe (the entire `child` object is already being cloned into `result` on line 229). 

```js
      // Special merging behaviour for args.
      for (var i = child.args.length ; i < this.args.length ; i++ ) {
        result.args[i] = this.args[i].clone().copyFrom(child.args[i]);
      }
```